src/Routers/FilesRouter.js

```diff
@@ -172,8 +172,22 @@ export class FilesRouter {
       }
     }
 
-    const base64 = req.body.toString('base64');
-    const file = new Parse.File(filename, { base64 }, contentType);
+    // If the request body is a buffer and it's size is greater than the V8 string size limit
+    // we need to use a Blob to avoid the V8 string size limit
+    const MAX_V8_STRING_SIZE_BYTES = 536_870_912;
+
+    let file;
+
+    if (
+      typeof Blob !== 'undefined' &&
+      Buffer.isBuffer(req.body) &&
+      req.body?.length >= MAX_V8_STRING_SIZE_BYTES
+    ) {
```

Comment:
```suggestion
    if (Buffer.isBuffer(req.body) && req.body?.length >= MAX_V8_STRING_SIZE_BYTES) {
```

See https://github.com/parse-community/parse-server/pull/9286#discussion_r1791863516